### PR TITLE
Map various things

### PR DESCRIPTION
**Pull request by copygirl**
See commits for details

### CODE REVIEW

**Comment by RX14** [comment]
Filthy mergepull commits. You should run `git config branch.16w33a.rebase true` to get rid of them when using `git pull`. Also should run `git config --global branch.autosetuprebase always` to do this automatically on all new branches/clones in all future repos.

**Comment by RX14** [comment]
Actually `git config --global --bool pull.rebase true` is probably the better config to change.

**Comment by copygirl** [comment]
Fixed my rebase screwup that would've probably rewritten this repo's history (which the conflict checker seems to be fine with?).

File: mappings/net/minecraft/entity/projectile/EntityArrowBase.mapping [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLASS none/aan net/minecraft/entity/projectile/EntityProjectile
+CLASS none/aan net/minecraft/entity/projectile/EntityArrowBase
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/projectile/EntityArrowBase.mapping:1]: Sure about that?

File: mappings/net/minecraft/potion/Potion.mapping [hunk 2]
```diff
@@ -1,7 +1,7 @@
 CLASS none/age net/minecraft/potion/Potion
 	FIELD a REGISTRY Lnone/cs;
-	FIELD b id Lnone/kp;
-	FIELD c maxRawId I
+	FIELD b WATER_TEX Lnone/kp;
```
Comment (asiekierka) [inline mappings/net/minecraft/potion/Potion.mapping:3]: This is a location, not the texture. WATER_TEX_LOC

File: mappings/net/minecraft/potion/PotionEffect.mapping [hunk 3]
```diff
@@ -24,8 +25,26 @@ CLASS none/ry net/minecraft/potion/PotionEffect
 	METHOD <init> (Lnone/ry;)V
 		ARG 0 other
 	METHOD a getEffectType ()Lnone/rx;
+	METHOD a toNbt (Lnone/dt;)Lnone/dt;
```
Comment (asiekierka) [inline mappings/net/minecraft/potion/PotionEffect.mapping]: No! serializeTag/deserializeTag

File: mappings/net/minecraft/potion/PotionEffect.mapping [hunk 4]
```diff
@@ -24,8 +25,26 @@ CLASS none/ry net/minecraft/potion/PotionEffect
...
+		ARG 0 update
+	METHOD b setPermanent (Z)V
+		ARG 0 value
 	METHOD c getAmplifier ()I
 	METHOD d isAmbient ()Z
 	METHOD e shouldShowParticles ()Z
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 0 other
 	METHOD f getTranslationKey ()Ljava/lang/String;
+	METHOD g getPermanent ()Z
```
Comment (asiekierka) [inline mappings/net/minecraft/potion/PotionEffect.mapping]: isPermanent/setPermanent for booleans...

**Comment by asiekierka** [comment]
Other than those, LGTM! Good work.

File: mappings/net/minecraft/entity/projectile/EntityArrowBase.mapping (hunk 1 above)
Comment (copygirl) [reply mappings/net/minecraft/entity/projectile/EntityArrowBase.mapping:1]: There's projectile classes (fireballs, snowballs, enderpearls, ...) that don't extend this class. `EntityArrowBase` is the base class for normal and specter arrows, and that's it.

File: mappings/net/minecraft/potion/Potion.mapping (hunk 2 above)
Comment (copygirl) [reply mappings/net/minecraft/potion/Potion.mapping:3]: I remember seeing a lot of texture locations defined as `BG_TEX` and similar, as such I thought that was the way to do it. I'd say the type already shows that it's a location.

Let's vote on it? :+1: for `WATER_TEX_LOC`, :-1: for `WATER_TEX`?

Comment (asiekierka) [reply mappings/net/minecraft/potion/Potion.mapping:3]: Eh, I suppose WATER_TEX is fine. It's not a big deal.

**Comment by copygirl** [comment]
Getting a hang of rebase. Fixed `isPermanent` and `de/serializeTag`.

**Comment by Darkhax** [comment]
Lets go to mars ^^

**Comment by asiekierka** [comment]
Yay!

**Pull request merged**